tools/azure-sdk-tools/devtools_testutils/azure_testcase.py

```diff
@@ -192,24 +192,55 @@ def create_basic_client(self, client_class, **kwargs):
             if _is_autorest_v3(client_class):
                 # Create azure-identity class
                 from azure.identity import ClientSecretCredential
-                credentials = ClientSecretCredential(
+                return ClientSecretCredential(
                     tenant_id=tenant_id,
                     client_id=client_id,
                     client_secret=secret
                 )
             else:
                 # Create msrestazure class
                 from msrestazure.azure_active_directory import ServicePrincipalCredentials
-                credentials = ServicePrincipalCredentials(
+                return ServicePrincipalCredentials(
                     tenant=tenant_id,
                     client_id=client_id,
                     secret=secret
                 )
         else:
             if _is_autorest_v3(client_class):
-                credentials = self.settings.get_azure_core_credentials()
+                return self.settings.get_azure_core_credentials()
             else:
-                credentials = self.settings.get_credentials()
+                return self.settings.get_credentials()
+
+    def get_async_credential(self, client_class):
```

Comment:
I would do a private `_get_real_credential` that does you if logic with the `is_async` , and expose it with two public API like you did then.
I really don't like duplicaion, for real :p. It makes maintenance a burden